src/ring_theory/subsemiring.lean

```diff
@@ -369,6 +425,32 @@ lemma closure_eq_of_le {s : set R} {t : subsemiring R} (h₁ : s ⊆ t) (h₂ :
   closure s = t :=
 le_antisymm (closure_le.2 h₁) h₂
 
+lemma to_subsemiring_coe :
+  (M.to_subsemiring : set R) = add_submonoid.closure (M : set R) := rfl
+
+lemma to_subsemiring_to_add_submonoid :
+  M.to_subsemiring.to_add_submonoid = add_submonoid.closure (M : set R) := rfl
+
+/-- The elements of the additive closure of a multiplicative submonoid `M` are exactly the
+elements of the subsemiring closure of `M`. -/
+lemma clss : (add_submonoid.closure (M : set R) : set R) = subsemiring.closure (M : set R) :=
```

Comment:
I think it might be more useful to show that `subsemiring.closure (M : set R) = M.to_subsemiring`, then this lemma falls out as just a bunch of coercions.

Reply:
I gave it a try, but I am still struggling.  I will attempt tomorrow to prove this, but any help to prove this version is welcome!

Reply:
Ok, I have not made progress on the proof, but I have understood your suggestion!  Once the lemma you state is proven, what is below is indeed a proof of `clss`:

```lean
lemma clss : (add_submonoid.closure (M : set R) : set R) = subsemiring.closure (M : set R) :=
by { rw eric's_clss, exact (submonoid.to_subsemiring_coe M).symm }
```

Reply:
Ok, now that I understood what was supposed to happen, it works!  I just pushed the final step!